runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -960,6 +960,79 @@ TR::Register *J9::X86::AMD64::TreeEvaluator::conditionalHelperEvaluator(TR::Node
 #endif
 
 
+TR::Register *J9::X86::TreeEvaluator::irdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT(TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads(), "iReadBarrierEvaluator");
```

Comment:
Hmm - this will be changed soon by the field watch work so I'm wondering if we should structure this more like if shouldGenerateReadBarriersForFieldLoads then do this sequence otherwise just treat it like an aloadi and other can add other meanings and uses?

Reply:
Good suggestion; however without seeing the actual change of field watch work, it is hard to image exactly what need be checked. If field watch work is merged before this PR, I would definitely updated this change accordingly; otherwise, I suggest to massage the code when field watch is delivered.